portfolio/src/main/webapp/script.js

```diff
@@ -265,3 +275,15 @@ function deleteAllComments() {
   // Delete all the comments in datastore.
   fetch('/delete-data', {method: 'POST'});
 }
+
+/**
+ * Fetch the blobstore URL and add it to the form on the frontend.
+ */
+function fetchBlobstoreUrlAndShowForm() {
+  fetch('/blobstore-upload-url').then(response => response.text())
+    .then((imageUploadUrl) => {
+      const messageForm = document.getElementById('upload-form');
+      messageForm.action = imageUploadUrl;
+      messageForm.classList.remove('hidden');
+    });
+}
```

Comment:
I think it's a good idea to think about the usability of these forms, in terms of design. In this photo, https://screenshot.googleplex.com/6YRahxgfb3e.png, there are two different submit buttons and three input sections. When I was testing this, I forgot what button worked for which input method. It would be good to have some text, or a more intuitive design that let the users know what is expected of them.

To be clear, I'm not expecting you to make any changes here. We're about to move onto the capstone and this design is just fine for what we were asking, so I don't want you to spend any more time here, but it's just a good thing to think about as we move into the capstone.

Reply:
Ok, that makes sense -- I agree the UI is very confusing, I was just focusing on the backend because I wanted to see if I could get the basic setup working in time.

I'll keep this in mind for the capstone project!